src/notebook/notebookKernel.ts

```diff
@@ -0,0 +1,206 @@
+import { Subject } from 'await-notify'
+import * as net from 'net'
+import * as path from 'path'
+import { uuid } from 'uuidv4'
+import * as vscode from 'vscode'
+import { createMessageConnection, MessageConnection, NotificationType, StreamMessageReader, StreamMessageWriter } from 'vscode-jsonrpc/node'
+import { getAbsEnvPath } from '../jlpkgenv'
+import { getJuliaExePath } from '../juliaexepath'
+import { generatePipeName } from '../utils'
+
+interface ExecutionRequest {
+    cell: vscode.NotebookCell
+    executionOrder: number
+}
+
+const notifyTypeDisplay = new NotificationType<{ mimetype: string, current_request_id: number, data: string }>('display')
```

Comment:
Are the messages supported here a subset of all messages supported by the Julia Jupyter kernel.
I'm trying to figure out if there are some missing messages or attributes.
E.g. when we get display data, do we have metadata for the images? (in Python, they support additional metadata for image mimetypes).

Similarly it seems we're missing a message for interrupt (which i believe is necessary for Widows), i could be mistaken & this might not be required at all.

Reply:
Yeah, so the interrupt message is missing, but that I think is the major one.

Once we move the Julia code over to reuse the REPL server code, we'll automatically get support for additional messages for example for the variables explorer.

For "rich" display types, right now we have one notification from the kernel to the extension that has a MIME type string, and a payload. I think that should be enough? Or do we need something beyond that?

Reply:
>  Or do we need something beyond that?

Not that i'm aware of.